lib/miq_automation_engine/engine/miq_ae_event.rb

```diff
@@ -121,28 +109,67 @@ def self.process_result(ae_result, aevent)
   rescue URI::InvalidURIError => err
   end
 
-  def self.automate_user
-    # TODO: Get the real userid to pass into Automate
-    user = User.where(:userid => "admin").first
-    raise "Admin user needed to raise events" unless user
-    user
-  end
-
-  def self.call_automate(obj_type, obj_id, attrs, instance_name,
-                         user, state = nil, message = nil)
+  def self.call_automate(obj, attrs, instance_name, message = nil)
+    user_id, group_id, tenant_id = automate_user_ids(obj)
+    attrs.merge!(
+      :user_id      => user_id,
+      :miq_group_id => group_id,
+      :tenant_id    => tenant_id
+    )
     args = {
-      :object_type      => obj_type,
-      :object_id        => obj_id,
+      :object_type      => obj.class.name,
+      :object_id        => obj.id,
       :attrs            => attrs,
       :instance_name    => instance_name,
-      :user_id          => user.id,
-      :miq_group_id     => user.current_group.id,
-      :tenant_id        => user.current_tenant.id,
-      :automate_message => message,
-      :state            => state
+      :user_id          => user_id,
+      :automate_message => message
     }
+
     MiqAeEngine.deliver(args)
   end
 
-  private_class_method :automate_user, :call_automate
+  def self.provider_event_target(event)
+    column = [:vm_or_template_id, :ems_id].find { |c| event.send(c).present? }
+    klass = column.to_s.split("_")[0..-2].join("_")
+    klass = "ext_management_system" if klass == "ems"
+
+    klass.classify.constantize.where(:id => event.send(column)).first
+  end
+
+  def self.miq_event_target(target)
+    case target
+    when VmOrTemplate, MiqServer
+      target
+    else
+      target.ext_management_system
+    end
+  end
+
+  def self.automate_user_ids(object)
+    # TODO: Get the real userid to pass into Automate
+    target = case object
+             when EmsEvent
+               provider_event_target(object)
+             else
+               miq_event_target(object)
+             end
+
+    if target.kind_of?(VmOrTemplate)
+      user  = target.evm_owner
```

Comment:
I am thinking in that case we would run with the group and admin.  Looks like we need to validate here that if a user is associated to the Vm that it is also in the group, otherwise user would need to be nil and we pick up super_admin below.